src/skins-qt/search-select.cc

```diff
@@ -0,0 +1,210 @@
+
+    auto label_album = new QLabel (_("Album:"));
+    m_entry_album = new QLineEdit;
+
+    auto label_artist = new QLabel (_("Artist:"));
+    m_entry_artist = new QLineEdit;
+
+    auto label_file_name = new QLabel (_("File Name:"));
+    m_entry_file_name = new QLineEdit;
+
+    m_checkbox_clearprevsel = new QCheckBox (_("Clear previous selection before searching"));
+    m_checkbox_autoenqueue = new QCheckBox (_("Automatically toggle queue for matching entries"));
+    m_checkbox_newplaylist = new QCheckBox (_("Create a new playlist with matching entries"));
+    m_checkbox_clearprevsel->setCheckState (Qt::Checked);
+
+    QObject::connect (m_checkbox_autoenqueue, & QAbstractButton::toggled, [this] () {
+        toggle_related_checkbox (m_checkbox_autoenqueue, m_checkbox_newplaylist);
+    });
+
+    QObject::connect (m_checkbox_newplaylist, & QAbstractButton::toggled, [this] () {
+        toggle_related_checkbox (m_checkbox_newplaylist, m_checkbox_autoenqueue);
+    });
+
+    auto search = new QPushButton (_("Search"));
+    auto cancel = new QPushButton (_("Cancel"));
+
+    auto buttonbox = new QDialogButtonBox;
+    buttonbox->addButton (search, QDialogButtonBox::AcceptRole);
+    buttonbox->addButton (cancel, QDialogButtonBox::RejectRole);
+
+    QObject::connect (buttonbox, & QDialogButtonBox::accepted, this, & SearchSelectDialog::search);
+    QObject::connect (buttonbox, & QDialogButtonBox::rejected, this, & QDialog::close);
+
+    auto hbox = audqt::make_hbox (nullptr);
+    hbox->addWidget (logo);
+    hbox->addWidget (help_text);
+
+    auto grid = new QGridLayout;
+    grid->addLayout (hbox, 0, 0, 1, 2, Qt::AlignLeft);
+    grid->addWidget (label_title, 1, 0, Qt::AlignRight);
+    grid->addWidget (m_entry_title, 1, 1);
+    grid->addWidget (label_album, 2, 0, Qt::AlignRight);
+    grid->addWidget (m_entry_album, 2, 1);
+    grid->addWidget (label_artist, 3, 0, Qt::AlignRight);
+    grid->addWidget (m_entry_artist, 3, 1);
+    grid->addWidget (label_file_name, 4, 0, Qt::AlignRight);
+    grid->addWidget (m_entry_file_name, 4, 1);
+    grid->addWidget (m_checkbox_clearprevsel, 5, 0, 1, 2);
+    grid->addWidget (m_checkbox_autoenqueue, 6, 0, 1, 2);
+    grid->addWidget (m_checkbox_newplaylist, 7, 0, 1, 2);
+
+    auto vbox = audqt::make_vbox (this);
+    vbox->addLayout (grid);
+    vbox->addWidget (buttonbox, 0, Qt::AlignBottom);
+}
+
+void SearchSelectDialog::toggle_related_checkbox (QCheckBox * caller, QCheckBox * related)
+{
+    if (caller->isChecked ())
+        related->setCheckState (Qt::Unchecked);
+}
+
+void SearchSelectDialog::copy_selected_to_new (Playlist playlist)
+{
+    int entries = playlist.n_entries ();
+    Index<PlaylistAddItem> items;
+
+    for (int entry = 0; entry < entries; entry ++)
+    {
+        if (playlist.entry_selected (entry))
+        {
+            items.append
+             (playlist.entry_filename (entry),
+              playlist.entry_tuple (entry, Playlist::NoWait));
```

Comment:
We could copy the decoder from the old playlist too (just an optimization):
```
            items.append
             (playlist.entry_filename (entry),
              playlist.entry_tuple (entry, Playlist::NoWait),
              playlist.entry_decoder (entry, Playlist::NoWait));
```

Reply:
Fixed, pushed as separate commit for skins and skins-qt.